### PR TITLE
Add relevant chat context when query the index

### PR DESCRIPTION
**Pull request by trducng**

### CODE REVIEW

File: libs/ktem/ktem/reasoning/simple.py [hunk 1]
```diff
@@ -253,6 +255,9 @@ def invoke(
             messages = []
             if self.system_prompt:
                 messages.append(SystemMessage(content=self.system_prompt))
+            for human, ai in history[-5:]:
```
Comment (lone17) [inline libs/ktem/ktem/reasoning/simple.py]: please make this an argument, a number of recent messages and/or the number of recent tokens

File: libs/ktem/ktem/reasoning/simple.py [hunk 2]
```diff
@@ -406,6 +419,45 @@ def run(self, question: str) -> Document:  # type: ignore
...
+                content="Summarize Cryptocurrency Market Dynamics from last year"
+            ),
+            HumanMessage(content="What are my health plans?"),
+            AIMessage(content="Show available health plans"),
+        ]
+        for human, ai in history[-3:]:
+            messages.append(HumanMessage(content=human))
+            messages.append(AIMessage(content=ai))
+
+        messages.append(HumanMessage(content=f"Generate search query for: {question}"))
```
Comment (lone17) [inline libs/ktem/ktem/reasoning/simple.py]: What is this hardcode ?

File: libs/ktem/ktem/reasoning/simple.py [hunk 3]
```diff
@@ -417,13 +469,24 @@ class Config:
...
     rewrite_pipeline: RewriteQuestionPipeline = RewriteQuestionPipeline.withx()
+    add_query_context: AddQueryContextPipeline = AddQueryContextPipeline.withx()
     use_rewrite: bool = False
 
-    def retrieve(self, message: str) -> tuple[list[RetrievedDocument], list[Document]]:
+    def retrieve(
+        self, message: str, history: list
+    ) -> tuple[list[RetrievedDocument], list[Document]]:
         """Retrieve the documents based on the message"""
+        if len(message) < 150:
```
Comment (lone17) [inline libs/ktem/ktem/reasoning/simple.py]: should be an argument

**Review by lone17 — CHANGES_REQUESTED** [review]

File: libs/ktem/ktem/reasoning/simple.py (hunk 2 above)
Comment (trducng) [reply libs/ktem/ktem/reasoning/simple.py]: @lone17 They are the few-shot examples.

Comment (lone17) [reply libs/ktem/ktem/reasoning/simple.py]: @trducng then the -3 should be an argument as well

File: libs/ktem/ktem/reasoning/simple.py [hunk 4]
```diff
@@ -311,6 +318,9 @@ async def ainvoke(  # type: ignore
             messages = []
             if self.system_prompt:
                 messages.append(SystemMessage(content=self.system_prompt))
+            for human, ai in history[-5:]:
```
Comment (lone17) [inline libs/ktem/ktem/reasoning/simple.py]: here as well

File: libs/ktem/ktem/reasoning/simple.py [hunk 5]
```diff
@@ -350,6 +361,9 @@ def stream(  # type: ignore
...
```
Comment (lone17) [inline libs/ktem/ktem/reasoning/simple.py]: also here

File: libs/ktem/ktem/reasoning/simple.py [hunk 6]
```diff
@@ -406,6 +420,49 @@ def run(self, question: str) -> Document:  # type: ignore
...
+                "the searching, return just the number 0."
+            ),
+            HumanMessage(content="How did crypto do last year?"),
+            AIMessage(
+                content="Summarize Cryptocurrency Market Dynamics from last year"
+            ),
+            HumanMessage(content="What are my health plans?"),
+            AIMessage(content="Show available health plans"),
+        ]
+        for human, ai in history[-3:]:
```
Comment (lone17) [inline libs/ktem/ktem/reasoning/simple.py]: same here

**Review by lone17 — CHANGES_REQUESTED** [review]

**Review by lone17 — APPROVED** [review]
lgtm

**Comment by lone17** [comment]
solve #46

**Pull request merged**